src/containerapp/azext_containerapp/java_component_decorator.py

```diff
@@ -214,3 +227,120 @@ def process_loaded_scg_route(self):
                 raise ValidationError(f'The "filters" field must be a list in route {route["id"]}. Please see https://aka.ms/gateway-for-spring-routes-yaml for a valid Gateway for Spring routes YAML spec.')
 
         return yaml_scg_routes.get('springCloudGatewayRoutes')
+
+
+class JavaComponentCreateDecorator(BaseJavaComponentDecorator):
+
+    def create(self):
+        try:
+            return self.client.create(
+                cmd=self.cmd, resource_group_name=self.get_argument_resource_group_name(),
+                environment_name=self.get_argument_environment_name(), name=self.get_argument_java_component_name(),
+                java_component_envelope=self.java_component_def,
+                no_wait=self.get_argument_no_wait())
+        except Exception as e:
+            stringErr = str(e)
+            if "JavaComponentsNotAllowedForSubscription" in stringErr:
+                raise CLIInternalError("Java Components operations are not allowed for the subscription, please use 'az feature register --namespace  Microsoft.App --name JavaComponentsPreview' to register this feature.")
```

Comment:
Not related to this PR, but we should remove the AFEC flag given that most of the Java components are now GA'ed.

Reply:
As we sync in teams, currently the AFEC flag does not take effect